src/ServiceControl/Recoverability/Retrying/RetryingManager.cs

```diff
@@ -37,7 +37,7 @@ public bool IsOperationInProgressFor(string requestId, RetryType retryType)
 
         public bool IsRetryInProgressFor(string requestId)
         {
-            return retryOperations.Keys.Where(key => key.EndsWith($"/{requestId}")).Any(key => retryOperations[key].IsInProgress());
```

Comment:
Accessing the `retryOperations` by key, even if `retryOperations` is a concurrent dict would not be safe here because there is no lock that makes sure the keys returned by `Keys` are still present when actually enumerating the returned collection. I've changed it to enumerating the values.

Reply:
I think you can leave it as is with the keys, `Keys` and `Values` both acquire all the locks, see https://referencesource.microsoft.com/#mscorlib/system/Collections/Concurrent/ConcurrentDictionary.cs,2036

`GetEnumerator` would be lock-free but might return dirty reads